### DISCUSSION TITLE
建议提供不同”输入法组“可以使用不同皮肤的功能

### DISCUSSION THREAD

**Discussion by xingwozhonghua126** [fcitx/fcitx5#393] [Ideas]
因为fcitx4适配输入法生态的时候碰到的主要问题之一，就是kimpanel问题，也就是输入法皮肤问题。
如果fcitx5输入法框架的”组“内部支持不同皮肤，各个厂商只需要做一个状态栏就好了。没必要关注候选词皮肤。在wayland下会有更好的体验。
有助于引导厂商专注于输入法引擎的优化。

**Comment by wengxt** [comment]
额……非要说的话，也不是不行，但最后不都还是得用 Fcitx 自己的主题引擎，这样真的可以吗 ´_>` ……

我并不是很明白你描述的需求细节，能再具体一些吗

**Reply by xingwozhonghua126** [reply]
fcitx5现在不是有输入法分组功能嘛，就是想在安装fcitx5搜狗的时候会新建一组输入法，并切换到搜狗输入法组，搜狗输入法组内部有搜狗英文，搜狗拼音，像fcitx5-chinese-addons一样。只是拿搜狗举例，也可能是迅飞，百度。

这样更符合fcitx的设计，不像是fcitx4或者windows那样，各个第三方输入法是复合输入法。

如果提供切换分组的时候，能切换不同皮肤，就可以区分使用的输入法了。

因为厂商更关注界面显示和商标，如果不显示出他们自己的商标，用户就无法区分当前使用的输入法哪个。

就比较难推fcitx5输入法生态。

**Reply by wengxt** [reply]
我是想理解这样一件事情，为什么要和分组绑定？当然，我觉得用分组来生成默认配置的话倒是一个还不错的主意。

单纯比如让使用搜狗的时候默认用搜狗设计的皮肤（尽管皮肤格式还是只能是 Fcitx 的那个格式）这样不行吗？皮肤没必要和分组绑定啊，和当前输入法绑定就行了呗？简单来说就是允许输入法声明「我想在用这个输入法的时候用这个皮肤」这样的机制是不是就行了？不需要加入和分组相关的逻辑让事情复杂化

**Reply by xingwozhonghua126** [reply]
`单纯比如让使用搜狗的时候默认用搜狗设计的皮肤（尽管皮肤格式还是只能是 Fcitx 的那个格式）这样不行吗？`

这样也是可以的。

`皮肤没必要和分组绑定啊，和当前输入法绑定就行了呗？`

是的，确实皮肤与输入法绑定相对于皮肤与输入法分组绑定会更好。

我的原有的想法是，一个输入法组对应一个厂商的输入法组感觉更完整。

还有一个点，想沟通一下，安装自动生成的输入法组不能被用户修改。

如果第三方输入法用分组来生成默认配置，但是这个分组还是可以被修改，就感觉有点混乱。

因为这个默认分组对应是完整的一个厂商的功能模块，当然模块的子集，比如“搜狗拼音”可以被加到其他组，但是不能移除搜狗组内部的搜狗拼音。
